src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationHandler.cs

```diff
@@ -321,6 +334,18 @@ protected override AuthenticationTicket AuthenticateCore()
                     _configuration = await Options.ConfigurationManager.GetConfigurationAsync(Context.RequestAborted);
                 }
 
+                // Redeeming authorization code for tokens
+                if (string.IsNullOrWhiteSpace(message.IdToken) && isCodeOnlyFlow)
```

Comment:
just because Options.ResponseType == CodeOnly, that does not mean the message is 'code' only. We need to determine the type of response from the actual message received. This is particularly important when a site is being attacked.